src/file_handle_cache.erl

```diff
@@ -581,6 +583,43 @@ info_keys() -> ?INFO_KEYS.
 info() -> info(?INFO_KEYS).
 info(Items) -> gen_server2:call(?SERVER, {info, Items}, infinity).
 
+clear_read_cache() ->
+    gen_server2:cast(?SERVER, clear_read_cache),
+    clear_vhost_read_cache(rabbit_vhost:list()).
+
+clear_vhost_read_cache([VHost | Rest]) ->
+    clear_queue_read_cache(rabbit_amqqueue:list(VHost)),
+    clear_vhost_read_cache(Rest);
+clear_vhost_read_cache([]) ->
+    ok.
+
+clear_queue_read_cache(
+  [#amqqueue{name = #resource{virtual_host = VHost, name = Name}, pid = Pid}
+  | Rest]) ->
+    rabbit_log:info(
+      "Clear read buffer cache for queue \"~s\" on vhost \"~s\"~n",
+      [Name, VHost]),
+    rabbit_amqqueue:run_backing_queue(Pid, rabbit_variable_queue,
+      fun(_, State) ->
+          clear_process_read_cache(),
```

Comment:
Why do you need to run this code in the context of the backing queue? `clear_process_read_cache/0` will call put in the context of the backing queue Pid if I'm not mistake, so that process dictionary will be used.